claim/claim.c

```diff
@@ -46,10 +48,29 @@ void claim_agent(char *claiming_arguments)
     char command_buffer[CLAIMING_COMMAND_LENGTH + 1];
     FILE *fp;
 
+    char *cloud_base_hostname = NULL; // Initializers are over-written but prevent gcc complaining about clobbering.
+    char *cloud_base_port = NULL;
+    char *cloud_base_url = config_get(CONFIG_SECTION_CLOUD, "cloud base url", "https://netdata.cloud");
+    if( aclk_decode_base_url(cloud_base_url, &cloud_base_hostname, &cloud_base_port))
+    {
+        error("Configuration error - cannot decode \"cloud base url\"");
+        return;
+    }
+
+    const char *proxy_str;
+    ACLK_PROXY_TYPE proxy_type;
+    char proxy_flag[CLAIMING_PROXY_LENGTH] = "-noproxy";
+
+    proxy_str = aclk_lws_wss_get_proxy_setting(&proxy_type);
+
+    if(proxy_type == PROXY_TYPE_SOCKS5)
+        snprintf(proxy_flag, CLAIMING_PROXY_LENGTH, "-proxy=\"%s\"", proxy_str);
+
     snprintfz(command_buffer,
               CLAIMING_COMMAND_LENGTH,
-              "exec netdata-claim.sh -hostname=%s -id=%s -url=%s %s",
-              netdata_configured_hostname,
+              "exec netdata-claim.sh %s -hostname=%s -id=%s -url=%s %s",
+              proxy_flag,
+              cloud_base_url,
```

Comment:
You're replacing the machine hostname with `cloud_base_url` and keeping `registry.cloud_base_url` as is. Whas this the intended use?

Reply:
should be:
```
    snprintfz(command_buffer,
              CLAIMING_COMMAND_LENGTH,
              "exec netdata-claim.sh %s -hostname=%s -id=%s -url=%s %s",
              proxy_flag,
              netdata_configured_hostname,
              localhost->machine_guid,
              cloud_base_url,
              claiming_arguments);
```

Reply:
I just need to check why I thought the test passed before I change the code.

Reply:
It was the PATH problem that @underhood mentioned in the summary. Test was failing to fail in the right way. Now I can see this in the log:
```
2020-03-17 08:32:43: netdata INFO  : MAIN : Waiting for claiming command to finish.
Token: ****************
Base URL: https://localhost:8443/
Id: 99fa2382-49b3-11ea-b9ff-448a5bd059c8
Rooms: 
Hostname: https://localhost:8443/
```
But I can also see this earlier in the sequence:
```
2020-03-17 08:32:42: netdata ERROR : MAIN : User rl=https://dev.private.netdata.cloud:443 -token=unAchqvrEB9nqo2LKM6788NdIMC6d
aH_6bPwikKfox3rlKm7xJb4dZFmlEASEK2IM1ZrYmCbT58fiXhjm4QTKltkzVPNnOHiVa4tJEIfv0JxM8rlzWiKXoK1E0lKgcxlLVx0NiQ -rooms=37925172-8ab
4-446d-b8bb-18538a3162be -proxy=socks5h://127.0.0.1:11081 -verbose is not present.
2020-03-17 08:32:42: netdata ERROR : MAIN : Cannot become user 'rl=https://dev.private.netdata.cloud:443 -token=unAchqvrEB9nqo
2LKM6788NdIMC6daH_6bPwikKfox3rlKm7xJb4dZFmlEASEK2IM1ZrYmCbT58fiXhjm4QTKltkzVPNnOHiVa4tJEIfv0JxM8rlzWiKXoK1E0lKgcxlLVx0NiQ -roo
ms=37925172-8ab4-446d-b8bb-18538a3162be -proxy=socks5h://127.0.0.1:11081 -verbose'. Continuing as we are.
```

Reply:
Tracing it through gdb I can see that the reason is that the configuration has been corrupted but it is not clear why yet:
```
[global]
	# run as user = rl=https://dev.private.netdata.cloud:443 -token=unAchqvrEB9nqo2LKM6788NdIMC6daH_6bPwikKfox3rlKm7xJb4dZFmlEASEK2IM1ZrYmCbT58fiXhjm4QTKltkzVPNnOHiVa4tJEIfv0JxM8rlzWiKXoK1E0lKgcxlLVx0NiQ -rooms=37925172-8ab4-446d-b8bb-18538a3162be -proxy=socks5h://127.0.0.1:11081 -verbose
```